src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -674,6 +687,11 @@ static int am_recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
     if ((uo_msg = MPIDI_OFI_am_claim_unordered_msg(fi_src_addr, next_seqno)) != NULL) {
         am_hdr = &uo_msg->am_hdr;
         orig_buf = am_hdr;
+#if NEEDS_STRICT_ALIGNMENT
+        /* alignment is ensured for this unordered message as it copies to a temporary buffer
+         * in MPIDI_OFI_am_enqueue_unordered_msg */
```

Comment:
This is good. Could you add another comment where this variable is first introduced (line 599) to explain the purpose of the variable, like "if this variable is off and the message contains extended header, the header needs to be copied out to access". -- As I typing this, I can't help feel that the name of this variable is still not right, but I'll let you decide.